static/compiler.js

```diff
@@ -106,47 +150,69 @@ function Compiler(domRoot, origFilters, windowLocalPrefix, onChangeCallback, lan
             break;
     }
 
+    // Set up the editor's window
+    // TODO : understand why [0] is mandatory here
     cppEditor = CodeMirror.fromTextArea(domRoot.find(".editor textarea")[0], {
         lineNumbers: true,
         matchBrackets: true,
         useCPP: true,
         mode: cmMode
     });
+
     // With reference to "fix typing '#' in emacs mode"
     // https://github.com/mattgodbolt/gcc-explorer/pull/131
     cppEditor.setOption("extraKeys", {
       "Alt-F": false
     });
     cppEditor.on("change", function () {
         if ($('.autocompile').hasClass('active')) {
-            onChange();
+            onEditorChange();
         }
     });
-    asmCodeMirror = CodeMirror.fromTextArea(domRoot.find(".asm textarea")[0], {
-        lineNumbers: true,
-        mode: "text/x-asm",
-        readOnly: true,
-        gutters: ['CodeMirror-linenumbers']
-    });
 
+    // Load/Save/Remove setting from the browser
     function getSetting(name) {
+        // source: http://stackoverflow.com/questions/9943220/how-to-delete-a-localstorage-item-when-the-browser-window-tab-is-closed
```

Comment:
I don't understand this comment: perhaps this should this be in `removeSetting` below?